BackendClassLibrary/DatabaseManager.cs

```diff
@@ -46,19 +55,28 @@ private DataTable runSelectQuery(string query)
         private bool runSqlInsertUpdateOrDeleteCommand(string query)
         {
             int rowsAffected;
-            try {
-                MySqlCommand cmd = new MySqlCommand(query, conn);
-                conn.Open();
-                rowsAffected = cmd.ExecuteNonQuery();
-                conn.Close();
-            } catch (MySqlException ex) {
-                Console.WriteLine("MYSQL ERROR OCCURED! ERROR MESSAGE: " + ex.Message);
-                return false;
-            }
             
+            lock (connectionLock)
+            {
+                try {
+                    MySqlCommand cmd = new MySqlCommand(query, conn);
+                    conn.Open();
+                    rowsAffected = cmd.ExecuteNonQuery();
+                    conn.Close();
+                } catch (Exception ex) {
+                    if (ex is MySqlException) {
+                        Console.WriteLine("MYSQL ERROR OCCURED! ERROR MESSAGE: " + ex.Message);
+                    } else {
+                        Console.WriteLine("ERROR OCCURED! ERROR MESSAGE: " + ex.Message);
+                    }
+                    conn.Close();
+                    return false;
+                }
+            }
+
             if (rowsAffected == 0) {
                 Console.WriteLine("The Command was Valid, but no Rows where affected: ");
-            } else if (rowsAffected < 0){
+            } else {
```

Comment:
This will trigger even if a command returns rows like we want. the -1 is returned if the command if for the wrong type